### ISSUE TITLE
v.1.0.6 UnhandledPromiseRejectionWarning: Error: out of memory

### ISSUE THREAD

**Issue by LhordeS** [HashLips/hashlips_art_engine#192]
When I run the program, it builds two iterations, then gives me the following error. I have replaced the layer images with my own png images, and changed all the folder names as per the tutorial. I have no idea what to look for in the directories listed in the error report. If someone could explain the problem to me in layman terms, that would be amazing! Cheers

(node:2244) UnhandledPromiseRejectionWarning: Error: out of memory
    at setSource (E:\NFT Projects\Operational NFT Collection\hashlips_art_engine-1.0.6_update\node_modules\canvas\lib\image.js:91:13)
    at Image.set (E:\NFT Projects\Operational NFT Collection\hashlips_art_engine-1.0.6_update\node_modules\canvas\lib\image.js:62:9)
    at E:\NFT Projects\Operational NFT Collection\hashlips_art_engine-1.0.6_update\node_modules\canvas\index.js:34:15
    at new Promise (<anonymous>)
    at loadImage (E:\NFT Projects\Operational NFT Collection\hashlips_art_engine-1.0.6_update\node_modules\canvas\index.js:23:10)
    at E:\NFT Projects\Operational NFT Collection\hashlips_art_engine-1.0.6_update\src\main.js:136:25
    at new Promise (<anonymous>)
    at loadLayerImg (E:\NFT Projects\Operational NFT Collection\hashlips_art_engine-1.0.6_update\src\main.js:135:10)
    at E:\NFT Projects\Operational NFT Collection\hashlips_art_engine-1.0.6_update\src\main.js:252:31
    at Array.forEach (<anonymous>)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:2244) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:2244) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by dhyeythumar** [comment]
Hi @LhordeS,
I guess you are using very big images in layers folder or some image got corrupted in between the operations which caused canvas library to fail.

**Comment by Spaksu** [comment]
You can use this command
`node --max-old-space-size=4096 index.js`
But be carefull with the size. Don't overload your pc.

**Comment by stugreenham** [comment]
Having the same issue - my mac is pretty powerful but it's not able to get past 1600 images before failing out. Not sure how to get it to finish a full batch of images. Any ideas?

**Comment by dhyeythumar** [comment]
Hi @stugreenham,
Did you try the solution provided by @Spaksu ?

**Comment by stugreenham** [comment]
> Hi @stugreenham,
> Did you try the solution provided by @Spaksu ?

I did - however this did not work. I managed to fix this though and weirdly it was one single file in a layer folder that was causing the issue. I compressed every single image to see if that helped and this one image wouldn't compress. I re-exported it and all was resolved.

**Comment by dhyeythumar** [comment]
Ok @stugreenham good to here the problem is solved.
I guess it there was an corrupted/damaged image. @LhordeS you should also check for this kind of images.

**Comment by dhyeythumar** [comment]
Ok @stugreenham good to hear that problem is solved.
I guess it there was an corrupted/damaged image. @LhordeS you should also check for this kind of images.

**Comment by LhordeS** [comment]
Thank you very much for your suggestions @dhyeythumar and @Spaksu. I actually had some very large background image files in one of the layer folders. I compressed them and was able to generate 1000 images no problem.
